modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators.Tests/ScriptPropertiesGeneratorTests.cs

```diff
@@ -0,0 +1,132 @@
+using System.Text;
+using Microsoft.CodeAnalysis.Text;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Godot.SourceGenerators.Tests;
+
+public class ScriptPropertiesGeneratorTestsTest : BaseTest<ScriptPropertiesGenerator>
+{
+    private readonly ITestOutputHelper _output;
+
+    public ScriptPropertiesGeneratorTestsTest(ITestOutputHelper output)
+    {
+        _output = output;
+
+        System.Console.SetOut(new OutputWriter(_output));
+    }
+
+    [Fact]
+    public async void EmptyTest()
+    {
+        var code = """
+        public partial class MyNode : Godot.Node
+        {
+        }
+        """;
+        var generated = """
+        using Godot;
+        using Godot.NativeInterop;
+
+        partial class MyNode
+        {
+        #pragma warning disable CS0109 // Disable warning about redundant 'new' keyword
+            /// <summary>
+            /// Cached StringNames for the properties and fields contained in this class, for fast lookup.
+            /// </summary>
+            public new class PropertyName : global::Godot.Node.PropertyName {
+            }
+        }
+
+        """;
+
+        await new CSharpSourceGeneratorVerifier<ScriptPropertiesGenerator>.Test
+        {
+            TestState =
+            {
+                Sources = { code },
+                GeneratedSources =
+                {
+                    (GeneratedFilename("MyNode"), SourceText.From(generated, Encoding.UTF8, SourceHashAlgorithm.Sha1)),
```

Comment:
Is it necessary to specify the hash algorithm? We don't ever specify in our generators so I assume they'll match with the default anyway.

Reply:
No, it's definitely not necessary.